docs/reference/platforms.rst

```diff
@@ -0,0 +1,259 @@
+Platforms
+=========
+
+Platforms
+---------
+
+The ``platforms`` keyword in a ``charmcraft.yaml`` determines where charms are
+built and where they run.
+
+For more information on how to build charms for specific bases and
+architectures, see the :doc:`Platforms how-to </howto/build-guides/select-platforms>`
+page.
+
+Standard notation
+~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+   platforms:
+     <platform-name>:
+       build-on:
+        - <my-arch>
+        - <your-arch>
+        ...
+       build-for:
+        - <our-arch>
+
+The platform name is an arbitrary string that describes the platform. The
+recommended platform name is the ``build-for`` arch.
+
+See :ref:`build-on <reference-build-on>` and
+:ref:`build-for <reference-build-for>` for information on the ``build-on``
+and ``build-for`` keywords.
+
+.. _reference-platforms-shorthand:
+
+Shorthand notation
+~~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+   platforms:
+     <arch>:
+
+Shorthand notation is a simple way to describe charms that build on and build
+for the same architecture. Shorthand notation requires the platform name to be
+a :ref:`valid debian architecture <reference-architectures>`.
+
+.. _reference-platforms-multi-base:
+
+Multi-base notation
+~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+   platforms:
+     <platform-name>:
+       build-on:
+        - <some-base>:<some-arch>
+        - <some-base>:<another-arch>
+        - <another-base>:<some-arch>
+        - <another-base>:<another-arch>
+       build-for:
+        - <base>:<arch>
```

Comment:
I think using `some-base`, `another-base`, and `base` may be misleading because craft-platforms will fail to parse a platform entry with multiple bases as described a few lines below with:

> Within each platform entry, the base must be the same for all ``build-on`` and ``build-for`` entries.

There will probably be an exception to this rule once we get closer to 26.04 and need to support `devel` bases.